doc/examples/denoise_ascm.py

```diff
@@ -0,0 +1,184 @@
+"""
+==============================================================
+Denoise images using Adaptive Soft Coefficient Matching (ASCM)
+==============================================================
+
+The adaptive soft coefficient matching (ASCM) as described in [Coupe11]_ is a
+improved extension of non-local means (NLMEANS) denoising. ASCM gives a better
+denoised images from two standard non-local means denoised versions of the
+original data with different degrees sharp feature preserved. Here, one
+denoised input is more "smooth" than the other (the easiest way to achieve this
+denoising is use ``non_local_means`` with two different patch radii).
+
+ASCM involves these basic steps
+
+* Computes wavelet decomposition of the noisy as well as denoised inputs
+
+* Combines the wavelets for the output image in a way that it takes it's
+  smoothness (low frequency components) from the input with larger smoothing,
+  and the sharp features (high frequency components) from the input with
+  less smoothing.
+
+This way ASCM gives us a well denoised output while preserving the sharpness
+of the image features, which is the primary goal in any denoising algorithm.
```

Comment:
I would remove ', which is the primary goal in any denoising algorithm' - we are stating the obvious!